Mathlib/MeasureTheory/Integral/Bochner.lean

```diff
@@ -1776,6 +1782,59 @@ theorem integral_mul_le_Lp_mul_Lq_of_nonneg {p q : ℝ} (hpq : p.IsConjugateExpo
 set_option linter.uppercaseLean3 false in
 #align measure_theory.integral_mul_le_Lp_mul_Lq_of_nonneg MeasureTheory.integral_mul_le_Lp_mul_Lq_of_nonneg
 
+theorem integral_countable' [Countable α] [MeasurableSingletonClass α] {μ : Measure α} {f : α → ℝ}
+  (hf : Integrable f μ):
+  ∫ a, f a ∂μ = ∑' a, f a * (μ {a}).toReal := by
```

Comment:
I wonder if this should be `(hf : Integrable f (μ.restrict s))` (aka `IntegrableOn`, but that notation isn’t available yet) instead, for a weaker precondition.

Maybe it helps to have something like
```
/-- If `s` is a countable set, then `μ.restrict s` is a sum of Dirac measures. -/
theorem restrict_eq_sum {s : Set α} (hs : Countable s) [MeasurableSingletonClass α] (μ : Measure α)
  : μ.restrict s = sum (s.indicator fun a => μ {a} • dirac a)
```
to match [`map_eq_sum`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Measure/Dirac.html#MeasureTheory.Measure.map_eq_sum), and then not go via the subtypes, but I am uncertain. I probably needs expert guidance.